### PR TITLE
feat(lts): add datasource to get all serach criteria

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add new datasource to get the list of the search criteria under specified log group.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add datasource to get all serach criteria.
2. add related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/lts TESTARGS='-run TestAccDataSourcSearchCriterias_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/lts -v -run TestAccDataSourcSearchCriterias_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourcSearchCriterias_basic
=== PAUSE TestAccDataSourcSearchCriterias_basic
=== CONT  TestAccDataSourcSearchCriterias_basic
--- PASS: TestAccDataSourcSearchCriterias_basic (51.82s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/lts       51.993s
```

### CODE REVIEW

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/lts TESTARGS='-run TestAccDataSourcSearchCriteria_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/lts -v -run TestAccDataSourcSearchCriteria_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourcSearchCriteria_basic
=== PAUSE TestAccDataSourcSearchCriteria_basic
=== CONT  TestAccDataSourcSearchCriteria_basic
--- PASS: TestAccDataSourcSearchCriteria_basic (47.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/lts       47.811s

File: docs/data-sources/lts_search_criteria.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+* `log_stream_name` - The name of the log stream.
+
+* `criteria` - The list of the search criteria under specified log stream.
+
+  The [criteria](#search_criteria_criteria_struct) structure is documented below.
+
+<a name="search_criteria_criteria_struct"></a>
+The `criteria` block supports:
+
+* `id` - The ID of the search criteria.
```
Comment (Lance52259) [inline docs/data-sources/lts_search_criteria.md]:
```suggestion
* `id` - The ID of the search criterion.
```

Comment (wuzhuanhong) [reply docs/data-sources/lts_search_criteria.md]: fixed

**Pull request merged**